src/caching_client.rs

```diff
@@ -73,7 +74,7 @@ impl CachingClient {
 
   #[inline]
   fn debuginfo_path(&self, build_id: &BuildId) -> PathBuf {
-    self.cache_dir.join(build_id.formatted()).join("debuginfo")
+    self.cache_dir.join(build_id.formatted().deref()).join("debuginfo")
```

Comment:
Mmm now that I am checking this again, still wrapping my head around `Cow`, I haven't seen any allocation code paths here, but not sure if they could happen...

Reply:
What I was referring to was the following in your v1 PR:
```rust
impl BuildId<'_> {
  pub fn formatted(&self) -> String {
    match self {
      BuildId::RawBytes(bytes) => format_build_id(bytes),
      BuildId::Formatted(string) => string.to_string(),     //  <------ allocates when it really doesn't have to
    }
  }
}
```

Reply:
Totally agreed, I think that with the current iteration there won't be allocations or copies anymore in this code path